papas/mergedclusterbuilder.py

```diff
@@ -67,14 +67,18 @@ def __init__(self, clusters, ruler, history_nodes = None):
     def _make_merged_clusters(self):
         #carry out the merging of linked clusters
         for subgraphids in self.subgraphs:
-            subgraphids.sort()
+            subgraphids.sort(reverse=True) #start with biggest clusters 
             first = None
             supercluster =None
             snode = None
+            totalenergy = 0.
+            #we want the merged cluster to have an identifier that matches its total energy
+            for elemid in subgraphids :
+                totalenergy += self.clusters[elemid].energy
```

Comment:
Yes, I realise this is not ideal. The issue is that without this additional total energy calculation the total energy is not known at the time that the MergedCluster unique identifier is made. I wanted the Identifier to know the total energy so that the Merged Clusters can be sorted by energy. Perhaps you have a better solution!